cogs/weightcog.py

```diff
@@ -557,6 +560,45 @@ async def export(self, ctx: commands.Context, user: discord.Member = None) -> No
                 file=discord.File(csv_file, filename=f"{user.display_name}_weight_data.csv"),
             )
 
+    @commands.command()
+    async def del_all_log(self, ctx: commands.Context, user: discord.Member = None) -> None:
+        """
+        Deletes all log data (CSV file) associated with a specified Discord user or the command invoker.
+
+        Parameters:
+            ctx (commands.Context): The context of the command.
+            user (discord.Member, optional): The Discord user for whom to delete the log data. Defaults to the command invoker.
+
+        Returns:
+            None: This function does not return any value.
+
+        Raises:
+            discord.ext.commands.MissingPermissions: If the command invoker lacks the necessary permissions.
+            discord.ext.commands.MissingRole: If the command invoker lacks the required bot-input role when specifying another user.
+
+        Example:
+            .del_all_log
+            .del_all_log user(Optional)
+        """
+        if user and (user != ctx.author) and not has_bot_input_perms(ctx):
+            await ctx.send(
+                "You don't have the bot-input role and are therefore not allowed to specify other"
```

Comment:
Nit: Line is 100 characters. So no need to put the await statement  on 4 lines.

Reply:
Could you elaborate on this.

Reply:
I believe he is referring to the `pyproject.toml` which overrides the default character limit

Reply:
Resolved